python/tach/interactive/modules.py

```diff
@@ -152,11 +158,19 @@ def _build_subtree(
                         continue
 
                     # Exclude patterns are relative to project root, and may include a trailing slash
-                    entry_path_for_matching = (
-                        f"{entry.relative_to(self.root.full_path)}/"
-                    )
+                    if exclude_paths is not None and use_regex_matching is None:
```

Comment:
We shouldn't break existing configs.

Ideally this pr simply provides to option for users to add the flag and turn it on if they want to use globs, but simply upgrading the package should not result in an error if you already have excludes.

Reply:
One thing we could do is validate that if you have `is_regex_matching` as `True`, is that all of the `exclude_paths` are valid regex. Similarly for if it's `False` and `globs`!

Reply:
It won't break the existing config, since the config is loaded with `ProjectConfig.with_derived_unset_fields`, and as the old configs won't have `use_regex_matching` field, it will default it automatically to True.
In short, this `if` condition is never meant to run for the end user. If it does, its our mistake somewhere

Reply:
Pattern validation is a nice idea, will add soon.